examples/onepanel_submitter.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import pyaml
+import os
+import json
+import zlib
+
+from couler.argo_submitter import ArgoSubmitter
+
+import onepanel.core.api
+from onepanel.core.api.rest import ApiException
+
+
+class OnepanelSubmitter(ArgoSubmitter):
+    """A submitter which submits a workflow to Onepanel"""
+
+    namespace = os.getenv('ONEPANEL_RESOURCE_NAMESPACE')
+    # username = 'admin'
+    # host = os.getenv('ONEPANEL_API_URL')
+
+    def __init__(self, username, workflow_name, host, token):
+        # NOTE: Need to take username, api_url here as well
+        # If token and api_url are not passed in, use the values from token file and environment variable
+        # If those don't exist, throw an exception
+        self.token = token
+        self.workflow_name = workflow_name
+        self.host = host
+        self.username = username
+        self.configuration = onepanel.core.api.Configuration(self.host)
+
+        logging.basicConfig(level=logging.INFO)
+        try:
+            with open('/var/run/secrets/kubernetes.io/serviceaccount/token') as f:
+                api_token = f.read()
+            self.configuration.api_key['authorization'] = api_token
+            self.configuration.api_key_prefix['authorization'] = 'Bearer'
+            logging.info('Onepanel configuration detected')
+
+        except Exception:
+            api_token = self._get_token(host, username, token)
+            self.configuration.api_key['authorization'] = api_token
+            self.configuration.api_key_prefix['authorization'] = 'Bearer'
+        except Exception as e:
+            print(e)
+        with onepanel.core.api.ApiClient(self.configuration) as api_client:
+            # Create an instance of the API class
+            self.api_instance = onepanel.core.api.WorkflowTemplateServiceApi(api_client)
+        logging.info('Initialized')
+
+    def _create_workflow_template(self, workflow_yaml, name):
+        manifest = self.template_to_argo(workflow_yaml)
+        if len(name) >= 20:
+            logging.info('Name must be 20 characters or less.')
+            return
+        # NOTE: `label` was added here
+        body = onepanel.core.api.WorkflowTemplate(
+            manifest=manifest,
+            name=name,
+            labels=[{'key': 'created-by', 'value': 'couler'}])
+        try:
+            workflow_template = self.api_instance.create_workflow_template(
+                self.namespace,
+                body
+            )
+            return workflow_template
+        except ApiException as e:
+            self._get_workflow_template(self.namespace, self.workflow_name, manifest, body)
```

Comment:
I would get the workflow first, if it exists, then create new version, otherwise create the workflow.